test/scripts/test_compact_ids.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+# Copyright (c) 2017 Google Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests correctness of opt pass tools/opt --compact-ids."""
+
+from __future__ import print_function
+
+import os.path
+import sys
+
+TEST_HOME_PATH = '/tmp/test_compact_ids'
```

Comment:
Oh, this isn't used anywhere?  Hm.. I'd recommend that TEST_HOME_PATH default to a directory under the current directory.  

Or use tempfile.mkdtemp

https://docs.python.org/2/library/tempfile.html 

But then you'll want an option to keep the directory if something fails.